source/backend/dal/Repositories/CompReqFinancialRepository.cs

```diff
@@ -42,9 +42,26 @@ public IEnumerable<PimsCompReqFinancial> GetAllByAcquisitionFileId(long acquisit
             {
                 query = query.Where(c => c.CompensationRequisition.IsDraft == false);
             }
+
             return query.AsNoTracking().ToArray();
         }
 
+        public IEnumerable<PimsCompReqFinancial> GetAllByLeaseFileId(long leaseFileId, bool? finalOnly)
+        {
+            this._user.ThrowIfNotAllAuthorized(Security.Permissions.CompensationRequisitionView);
```

Comment:
nit: can you just use ThrowIfNotAuthorized here (there is only one permission).

Reply:
Updated